cdap-common/src/main/java/co/cask/cdap/common/utils/TimeMathParser.java

```diff
@@ -49,8 +49,11 @@ private static long convertToSeconds(String op, long num, String unitStr) {
       seconds = TimeUnit.HOURS.toSeconds(num);
     } else if ("d".equals(unitStr)) {
       seconds = TimeUnit.DAYS.toSeconds(num);
+    } else if ("ms".equals(unitStr)) {
+      seconds = TimeUnit.MILLISECONDS.toSeconds(num);
```

Comment:
this is incorrect,  if a user specifies, (now - 500ms), we round it to the last one second and include additional stream events. you should convert everything to milliseconds resolution, for metrics handler the resolution is still seconds, while the stream handler its in milliseconds, you have to handle both of them appropriately.